vagrant/_commons.sh

```diff
@@ -12,29 +12,131 @@ set -o pipefail
 set -o xtrace
 set -o nounset
 
+# Kubespray configuration values
+kubespray_folder=/opt/kubespray
+
+# uninstall_k8s() - Uninstall Kuberentes deployment
+function uninstall_k8s {
+    ansible-playbook -vvv -i ./inventory/hosts.ini $kubespray_folder/reset.yml --become
+}
+
+# _install_docker() - Download and install docker-engine
+function _install_docker {
+    if docker version &>/dev/null; then
+        return
+    fi
+
+    source /etc/os-release || source /usr/lib/os-release
+    case ${ID,,} in
+        clear-linux-os)
+            sudo -E swupd bundle-add containers-basic
+            sudo systemctl unmask docker.service
+        ;;
+        *)
+            curl -fsSL https://get.docker.com/ | sh
+        ;;
+    esac
+
+    sudo mkdir -p /etc/systemd/system/docker.service.d
+    if [ -n "$HTTP_PROXY" ]; then
+        echo "[Service]" | sudo tee /etc/systemd/system/docker.service.d/http-proxy.conf
+        echo "Environment=\"HTTP_PROXY=$HTTP_PROXY\"" | sudo tee --append /etc/systemd/system/docker.service.d/http-proxy.conf
+    fi
+    if [ -n "$HTTPS_PROXY" ]; then
+        echo "[Service]" | sudo tee /etc/systemd/system/docker.service.d/https-proxy.conf
+        echo "Environment=\"HTTPS_PROXY=$HTTPS_PROXY\"" | sudo tee --append /etc/systemd/system/docker.service.d/https-proxy.conf
+    fi
+    if [ -n "$NO_PROXY" ]; then
+        echo "[Service]" | sudo tee /etc/systemd/system/docker.service.d/no-proxy.conf
+        echo "Environment=\"NO_PROXY=$NO_PROXY\"" | sudo tee --append /etc/systemd/system/docker.service.d/no-proxy.conf
+    fi
+    if [[ "${CONTAINER_RUNTIME:-runc}" == "kata" ]]; then
```

Comment:
I declared `docker` as default value for `CONTAINER_MANAGER` and `runc` as `CONTAINER_RUNTIME`, from my perspective `Katacontainers` and `CRI-O` are still under development, isn't it?

Reply:
I'd say when setting up k8s, we should use CRI-O/containerd as ```CONTAINER_MANAGER``` (CRI-O is OK to start with) but not Docker anymore.

With proper (and new enough) CRI runtime used, ```CONTAINER_RUNTIME``` in these scripts gets void because the pods use ```RuntimeClassName``` to choose their preferred runtime. There could be a boolean flag to check whether kata-deploy should be run.

Reply:
To get this PR merged, we can keep ```CONTAINER_MANAGER``` as it is but let's not make separation in these scripts about kata vs runc.

Later let's change the scripts so that the k8s setup uses a CRI runtime.

Reply:
Okay, let me create a commit that removes it

Reply:
Removed on  d86583c